tests/pyyaml_test.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of HEPData.
+# Copyright (C) 2021 CERN.
+#
+# HEPData is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# HEPData is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with HEPData; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+import yaml
+try:
+    from yaml import CSafeLoader as Loader
+except ImportError:  # pragma: no cover
+    from yaml import SafeLoader as Loader  # pragma: no cover
```

Comment:
Does the test pass if `SafeLoader` is used instead of `CSafeLoader`? Perhaps we should check both - or if `SafeLoader` doesn't work, should we reconsider whether we should use it in the main code?

Reply:
Thanks for the comment.  I've added a test that `SafeLoader` raises a `yaml.scanner.ScannerError`.  This bug might be fixed in a future PyYAML version, then the test would fail if we upgraded the PyYAML version in `requirements.txt`.  But now that PyYAML v5.4.1 includes the LibYAML extension in the Python wheel, I don't see a reason to use pure-Python PyYAML (which is also slower), so I've removed usage of `SafeLoader` etc. throughout the code.